src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/DataTableCaption.test.tsx

```diff
@@ -0,0 +1,378 @@
+      ],
+    };
+    const { container } = render(
+      <DataTableCaption id="DataTableCaption" {...props} />,
+    );
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <strong
+          id="DataTableCaption"
+        >
+          Table showing Authorised absence rate for 'Absence by characteristic' from 'Pupil absence in schools in England' in Adur, Allerdale, Barking and Dagenham, Barnet and England for 2015/16
+        </strong>
+      </div>
+    `);
+  });
+
+  test('with more than 10 LocationFilters', () => {
+    const props = {
+      ...generateTableTitleBasicInput,
+      locations: [
+        ...generateTableTitleBasicInput.locations,
+        new LocationFilter({
+          value: 'E09000004',
+          label: 'Eleven',
+          level: 'localAuthority',
+        }),
+        new LocationFilter({
+          value: 'E09000002',
+          label: 'Ten',
+          level: 'localAuthority',
+        }),
+        new LocationFilter({
+          value: 'E09000003',
+          label: 'Nine',
+          level: 'localAuthority',
+        }),
+        new LocationFilter({
+          value: 'E07000223',
+          label: 'Eight',
+          level: 'localAuthorityDistrict',
+        }),
+        new LocationFilter({
+          value: 'E07000026',
+          label: 'Seven',
+          level: 'localAuthorityDistrict',
+        }),
+        new LocationFilter({
+          value: 'E07000027',
+          label: 'One',
+          level: 'localAuthorityDistrict',
+        }),
+        new LocationFilter({
+          value: 'E07000028',
+          label: 'Two',
+          level: 'localAuthorityDistrict',
+        }),
+        new LocationFilter({
+          value: 'E07000029',
+          label: 'Three',
+          level: 'localAuthorityDistrict',
+        }),
+        new LocationFilter({
+          value: 'E07000030',
+          label: 'Four',
+          level: 'localAuthorityDistrict',
+        }),
+        new LocationFilter({
+          value: 'E07000031',
+          label: 'Five',
+          level: 'localAuthorityDistrict',
+        }),
+        new LocationFilter({
+          value: 'E07000032',
+          label: 'Six',
+          level: 'localAuthorityDistrict',
+        }),
+      ],
+    };
+    const { container } = render(
+      <DataTableCaption id="DataTableCaption" {...props} />,
+    );
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <strong
+          id="DataTableCaption"
+        >
+          Table showing Authorised absence rate for 'Absence by characteristic' from 'Pupil absence in schools in England' in Eight, Eleven, England, Five, Four, Nine, One, Seven, Six, Ten and 2 other locations... for 2015/16
+        </strong>
+        <button
+          class="button govuk-!-display-block govuk-!-margin-top-2"
+          type="button"
+        >
+          View full table title
+        </button>
+      </div>
+    `);
+
+    fireEvent.click(screen.getByText('View full table title'));
```

Comment:
Use the `ByRole` selector: `screen.getByRole('button', { name: 'View full table title' })`